pkg/util/server/error.go

```diff
@@ -60,10 +62,22 @@ func ClientHTTPStatusAndError(err error) (int, error) {
 		return http.StatusBadRequest, err
 	case errors.Is(err, user.ErrNoOrgID):
 		return http.StatusBadRequest, err
+	case isRPC && (int(s.Code())/100 == 4 || int(s.Code())/100 == 5):
+		return int(s.Code()), errors.New(s.Message())
 	default:
 		if grpcErr, ok := httpgrpc.HTTPResponseFromError(err); ok {
 			return int(grpcErr.Code), errors.New(string(grpcErr.Body))
 		}
 		return http.StatusInternalServerError, err
 	}
 }
+
+// WrapError wraps an error in a protobuf status.
+func WrapError(err error) *rpc.Status {
+	code, err := ClientHTTPStatusAndError(err)
```

Comment:
Hm... This is strange. So, in the case of a `Status` error, i.e., an error recognized by `status.FromError()` or `grpcutil.ErrorToStatus()`, you first split those errors into `<code, error message>` pairs (inside `ClientHTTPStatusAndError()`), and then you create another `Status` error containing exactly the same `code` and `error message`

Reply:
Yes. We probably should check if we have a status or not. Good point!